xdsl/dialects/onnx.py

```diff
@@ -0,0 +1,93 @@
+from __future__ import annotations
+
+from abc import ABC
+
+from xdsl.dialects.builtin import AnyTensorType, SSAValue
+from xdsl.ir import (
+    Attribute,
+    Dialect,
+    OpResult,
+)
+from xdsl.irdl import (
+    IRDLOperation,
+    irdl_op_definition,
+    operand_def,
+    result_def,
+)
+from xdsl.utils.exceptions import VerifyException
+
+
+class ElementwiseBinOpBase(IRDLOperation, ABC):
+    """Base class for element-wise binary operations on tensors with Numpy-style broadcasting."""
+
+    lhs = operand_def()
+    rhs = operand_def()
+    res: OpResult = result_def(AnyTensorType)
+    assembly_format = "`(` $lhs `,` $rhs `)` attr-dict `:` `(` type($lhs) `,` type($rhs) `)` `->` type($res)"
+
+    def __init__(
+        self,
+        lhs: SSAValue,
+        rhs: SSAValue,
+        attributes: dict[str, Attribute] = {},
```

Comment:
Just spotted this, could you please remove argument? We try to keep our APIs fully typed, and then add dynamic attributes separately. Now that I think about it, adding an attributes argument might be a good idea in general, but I'd rather keep the API consistent for now. For the custom parser, we have a number of examples of adding attributes after the operation has been created in the codebase